component/main.jsonnet

```diff
@@ -2,9 +2,71 @@
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
 local inv = kap.inventory();
-// The hiera parameters for the component
 local params = inv.parameters.appcat_service_postgresql;
+local appName = 'appcat-service-postgresql';
 
-// Define outputs below
+local serviceAccount = kube.ServiceAccount(appName) {
+  metadata+: { namespace: params.namespace },
+};
+
+local clusterRole = kube.ClusterRole(appName) {
+  rules: [
+    {
+      apiGroups: [ 'postgresql.appcat.vshn.io' ],
+      resources: [ 'postgresqlstandalones', 'postgresqlstandaloneconfigs' ],
+      verbs: [ 'get', 'patch', 'list' ],
+    },
+    {
+      apiGroups: [ '*' ],
+      resources: [ 'statefulsets', 'configmaps', 'services', 'pods', 'secrets' ],
+      verbs: [ '*' ],
+    },
+    {
+      apiGroups: [ 'extensions' ],
+      resources: [ 'ingress' ],
+      verbs: [ '*' ],
+    },
+  ],
+};
+
+local clusterRoleBinding = kube.ClusterRoleBinding(appName) {
+  subjects_: [ serviceAccount ],
+  roleRef_: clusterRole,
+};
+
+local deployment = kube.Deployment(appName) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+  spec+: {
+    template+: {
+      spec+: {
+        containers_+: {
+          appName: kube.Container(appName) {
+            image: params.registry + '/' + params.image + ':' + params.tag,
+            imagePullPolicy: 'Always',
+            ports_+: {
+              'service-postgresql-port': {
+                containerPort: 6380,
+              },
+            },
+          },
+        },
+        securityContext: {
+          runAsUser: 10001,
+        },
+        serviceAccountName: serviceAccount.metadata.name,
+      },
+    },
+  },
+};
+
+local service = kube.Service(appName) {
+  target_pod: deployment.spec.template,
+};
```

Comment:
We should reuse the chart in https://github.com/vshn/appcat-service-postgresql/tree/master/chart, so that we don't have to duplicate lots of the deployment. See also https://syn.tools/syn/explanations/commodore-components/helm-charts.html.
Please try out if something like this works for the Chart URL: `git+https://github.com/vshn/appcat-service-postgresql@charts?ref=master`
We might consider moving the chart to https://github.com/appuio/charts if we can't install charts directly from Git.

Reply:
Or alternatively, we make this repository also a Helm chart repo using GitHub Pages with a single chart if it doesn't work.
This needs some automation though.

Or completely other idea: We integrate the component directly into https://github.com/vshn/appcat-service-postgresql, removing the chart altogether. Might need some more thoughts and discussions and maybe for later.

Reply:
{quote}
Please try out if something like this works for the Chart URL: git+https://github.com/vshn/appcat-service-postgresql@charts?ref=master
{quote}
Seems like the URL does not work. Can we move the chart to a valid repository? Otherwise I will use a dump source.

{quote}
We might consider moving the chart to https://github.com/appuio/charts if we can't install charts directly from Git.
Or alternatively, we make this repository also a Helm chart repo using GitHub Pages with a single chart if it doesn't work.
This needs some automation though.
{quote}
I would not put it into appuio charts or any single chart repo. I believe we need a separate chart repo for our services not just postgresql.

Reply:
Since we lack the automation for deploying charts in a monorepo I suggest the following:
- Copy the current chart to appuio/charts.
- Create a follow-up issue/ticket to engineer this automation (similar to https://github.com/k8up-io/k8up/issues/649)
- Once we have this automation, delete chart in appuio/charts

Another charts repo just for all appcat services doesn't really make sense as there's little benefit doing so (might as well use the appuio/charts repo). Developing with feature branches that require changes in the deployment becomes much easier in a monorepo than in an external charts repo.